relay-event-normalization/src/transactions/processor.rs

```diff
@@ -163,14 +179,61 @@ impl Processor for TransactionsProcessor<'_> {
         _meta: &mut Meta,
         state: &ProcessingState<'_>,
     ) -> ProcessingResult {
-        span.op.get_or_insert_with(|| "default".to_owned());
-
+        if span.op.value().is_none() {
+            *span.op.value_mut() = Some(self.span_op_defaults.infer(span));
+        }
         span.process_child_values(self, state)?;
 
         Ok(())
     }
 }
 
+/// Rules used to infer `span.op` from other span fields.
+#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
+pub struct SpanOpDefaults {
+    /// List of rules to apply. First match wins.
+    pub rules: Vec<SpanOpDefaultRule>,
+}
+
+impl SpanOpDefaults {
+    /// Gets a borrowed version of this config.
+    pub fn borrow(&self) -> BorrowedSpanOpDefaults {
+        BorrowedSpanOpDefaults {
+            rules: self.rules.as_slice(),
+        }
+    }
+}
+
+/// Borrowed version of [`SpanOpDefaults`].
+#[derive(Clone, Copy, Debug, Default)]
+pub struct BorrowedSpanOpDefaults<'a> {
+    rules: &'a [SpanOpDefaultRule],
+}
+
+impl BorrowedSpanOpDefaults<'_> {
+    /// Infer the span op from a set of rules.
+    ///
+    /// The first matching rule determines the span op.
+    /// If no rule matches, `"default"` is returned.
+    fn infer(&self, span: &Span) -> String {
+        for rule in self.rules {
+            if rule.condition.matches(span) {
+                return rule.value.clone();
+            }
+        }
+        "default".to_owned()
```

Comment:
Untested:


```suggestion
    self.rules.iter()
        .find(|rule| rule.condition.matches(span))
        .cloned()
        .unwrap_or("default".to_owned())
```

Reply:
I kind of like the for loop, even though it takes one more line.